### PR TITLE
WIP: Try to automatically reconnect when connection is lost

### PR DESCRIPTION
**Pull request by EmilioPeJu**
There is a parameter to disable this behaviour if needed.

For sake of simplicity, the camera parameter resynchronization is left to the user, in my specific application, I rely on the UserSetSave feature.

### CODE REVIEW

File: aravisApp/src/ADAravis.cpp [hunk 1]
```diff
@@ -356,6 +358,7 @@ ADAravis::ADAravis(const char *portName, const char *cameraName, int enableCachi
     createParam("ARAVIS_SHIFT_BITS",     asynParamInt32,   &AravisShiftBits);
     createParam("ARAVIS_CONNECTION",     asynParamInt32,   &AravisConnection);
     createParam("ARAVIS_RESET",          asynParamInt32,   &AravisReset);
+    createParam("ARAVIS_AUTO_CONNECT", asynParamInt32, &AravisAutoConnect);
```
Comment (ericonr) [inline aravisApp/src/ADAravis.cpp]: Nit: not following whitespace formatting.

File: aravisApp/src/ADAravis.cpp [hunk 2]
```diff
@@ -670,6 +677,22 @@ asynStatus ADAravis::allocBuffer() {
...
 }
 
+void ADAravis::reconnectIfNeeded() {
+    int autoConnect;
+    if (this->connectionValid == 0) {
+        getIntegerParam(AravisAutoConnect, &autoConnect);
+        if (autoConnect) {
+            /* lock to make sure the main thread doesn't connect at the
+             * same time via the reset PV */
+            this->lock();
```
Comment (ericonr) [inline aravisApp/src/ADAravis.cpp]: Isn't it unsafe to call `getIntegerParam` without the lock being held? A PV write could have triggered a write to the paramList at the same time.

File: aravisApp/src/ADAravis.cpp (hunk 1 above)
Comment (EmilioPeJu) [reply aravisApp/src/ADAravis.cpp]: Done!

File: aravisApp/src/ADAravis.cpp (hunk 2 above)
Comment (EmilioPeJu) [reply aravisApp/src/ADAravis.cpp]: Thanks for catching that

**Comment by EmilioPeJu** [comment]
Thanks for reviewing, this PR needs to be changed to do a different approach using the standard asyn auto-connect mechanism, I've marked it as WIP until it's implemented

**Comment by EmilioPeJu** [comment]
Closing because there is no demand for this feature, specially because here(at DLS), we rely on autosave to set the right camera parameters, and this requires the IOC to be restarted (at least without implementing some kind of connect hook mechanism).

**Comment by MarkRivers** [comment]
This feature was recently implemented and tested with ADVimba.  It uses a callback from the SDK when any camera connects or disconnects.  If it the camera run by that driver then it closes or opens the connection.  The OPI shows the connect/disconnect status.  The autosave configMenu feature can be used to manually push the EPICS state down to the camera on reconnect.

This might be implemented in ADAravis by detecting when the camera becomes reachable/unreachable in the poller.

**Pull request closed without merging**